src/Workspaces/CoreTestUtilities/Remote/InProcRemoteHostClientProvider.cs

```diff
@@ -55,15 +56,27 @@ public InProcRemoteHostClientProvider(HostWorkspaceServices services)
         {
             _services = services;
 
+            _lazyManager = new Lazy<WorkspaceManager>(() => new WorkspaceManager(RemoteAssetStorage ?? new SolutionAssetCache(), AdditionalRemoteParts));
             _lazyClient = new AsyncLazy<RemoteHostClient>(
                 cancellationToken => InProcRemoteHostClient.CreateAsync(
                     _services,
-                    new RemoteHostTestData(
-                        new WorkspaceManager(RemoteAssetStorage ?? new SolutionAssetCache(), AdditionalRemoteParts),
-                        isInProc: true)),
+                    new RemoteHostTestData(_lazyManager.Value, isInProc: true)),
                 cacheResult: true);
         }
 
+        public void Dispose()
+        {
+            // Dispose the remote workspace when the owning host workspace is disposed
+            if (_lazyManager.IsValueCreated)
+            {
+                var manager = _lazyManager.Value;
```

Comment:
why not just call .Dipose on the manager and allow it to decide what to do.

Reply:
`RemoteWorkspaceManager` is production code, but here we are in test code. I could make `WorkspaceManager` above disposable, but I'm not sure that would change anything.